tests/components/edl21/test_config_flow.py

```diff
@@ -0,0 +1,86 @@
+"""Tests for the EDL21 config flow."""
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.components.edl21.sensor import DOMAIN, CONF_SERIAL_PORT
+from homeassistant.components.edl21.config_flow import CONF_MANUAL_PATH
+from homeassistant.components.hassio import HassioServiceInfo
+from homeassistant.config_entries import SOURCE_USER
+from homeassistant.const import CONF_NAME
+from homeassistant.core import HomeAssistant
+import homeassistant.components.edl21
+from homeassistant.const import CONF_SOURCE
+
+from unittest.mock import AsyncMock, MagicMock, patch, sentinel
+
+from tests.common import MockConfigEntry
+
+import pytest
+import serial.tools.list_ports
+
+
+def com_port():
+    """Mock of a serial port."""
+    port = serial.tools.list_ports_common.ListPortInfo("/dev/ttyUSB1234")
+    port.serial_number = "1234"
+    port.manufacturer = "Virtual serial port"
+    port.device = "/dev/ttyUSB1"
+    port.description = "Some serial port"
+
+    return port
+
+
+@patch("serial.tools.list_ports.comports", MagicMock(return_value=[]))
+async def test_no_port_config_form(hass: HomeAssistant) -> None:
+    """Test that the setup form is served."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={CONF_SOURCE: SOURCE_USER}
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "pick_manual"
```

Comment:
Please continue the flow from here until the end (implement a start to finish test, instead of smaller tests for each bit). This ensures it will work altogether.